server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertMultiTabletPlan.java

```diff
@@ -384,4 +388,59 @@ public void unsetIsExecuted(int i) {
       results.remove(i);
     }
   }
+
+  public int getDifferentStorageGroupsCount() {
+    if (differentStorageGroupsCount == null) {
+      Set<String> insertPlanSGSet = new HashSet<>();
+      for (InsertTabletPlan insertTabletPlan : insertTabletPlanList) {
+        IStorageGroupMNode storageGroupMNode = null;
+        try {
+          storageGroupMNode =
+              IoTDB.metaManager.getStorageGroupNodeByPath(insertTabletPlan.getDeviceId());
+          insertPlanSGSet.add(storageGroupMNode.getFullPath());
+        } catch (MetadataException ignored) {
+        }
+      }
+      differentStorageGroupsCount = insertPlanSGSet.size();
+    }
+    return differentStorageGroupsCount;
+  }
+
+  public boolean isEnableMultiThreading() {
+    // After testing, we can find that when there are 10 columns , the thread pool speed will exceed
+    // the serial speed, so we set the threshold to 10.
```

Comment:
```suggestion
```
Explain why need the InsertMultiTabletEnableMultithreadingColumnThreshold configuration parameter here.